tests/data/CMakeLists.txt

```diff
@@ -0,0 +1,35 @@
+message(STATUS "Generating test data AppImage")
+
+# Get appimagetool
+set(APPIMAGETOOL_BIN ${CMAKE_CURRENT_BINARY_DIR}/appimagetool)
```

Comment:
This change is completely unrelated to the actual functionality you added. I am not a fan of this ***at all***. It's a ***horrible*** concept. You should not rely on unstable moving targets in unit tests! NEVER!

Reply:
This change is required to avoid adding **another** AppImage binary to the repository. 

I understand your concern about depending on unstable releases so the latest stable release is being used instead.

Reply:
These external dependencies you mention however are all pinned in some way. *You* introduce a dependency on an unstable system. The case where such a dependency might be broken and tests start failing at random is unacceptable. The reason is that unit tests should not just fail randomly, that makes debugging extremely hard. Unit tests should be as self-contained as possible, and even if there's external dependencies, these must be pinned so you can at any time restore a given point in the development history. Otherwise, things like reproducibility of bugs aren't provided any more. That's really a no-go.

If you find a way to pin that stuff somehow, your change is more viable.

Reply:
It's already done, now the code only downloads the 12th release of appimagetool so a random broken build will not break the tests cases.

Reply:
If you open an issue to change this to something less download-stuff-from-Internet, then this might be viable. The problem is that this again breaks builds on e.g., OBS.
CC @probonopd.

Reply:
At build time nothing must be downloaded because that won't work on OBS.

Reply:
As we already download a lot of stuff from the internet I thought that it was safe to do it. I guess that we already have an older version of appimagetool on the repo. It should be fine to use it.

Reply:
Wait, there is an appimagetool in the repo but only for i386 and x86_64. It would not be possible to build the required AppImage in other platforms.

Reply:
Right now I can only see two options:
1. do what ever is required to ensure the build succeed on OBS
2. include the new AppImage in the repo

If you can imagine something else, please let me know.

Reply:
The problem is I'm looking into better compliance and dependency control, and most of what we have is a big nightmare. I'll think about it.

Reply:
> after hearing the words "dependency control" a little bird flies near @TheAssassin saying "conan", "conan", "conan" 

Just a little weekend joke :smile:

Reply:
That also doesn't solve our principal problem, which is that we have no means of generating a complete list of all dependencies, and check whether we're complying with all of those. Not at all. It might contribute, but it also has lots of drawbacks, e.g., for OBS.

Reply:
It does resolve this *generating a complete list of all dependencies*. About OBS, I have to google a bit :)

Reply:
What a naive claim. As said, it might contribute, but:

a) we still have to check *all of our own* packages for conan for compliancy
b) as soon as you use anything third party, you need to look into those *by hand* to check whether all licenses have been declared correctly
c) there will always be non-conan dependencies

Don't be so naive to think license compliance was that easy. That list will not be complete, though it might come very close.

Reply:
Well, if it contributes to the solution it's not a naive idea. As you know I already use conan as part of AppImageService and is working really well. 

a) I already have done that part, had to apply some small patches but nothing outstanding. Final apps like AIL doesn't need to be turned into conan packages.

b) Licenses are included in conan packages, so it's not complicated at all to check it. I wonder if there is a package manager that checks licensing issues.

c) Yes, the linux kernel and compilers. Besides those is possible to have a conan package for almost everithing. There are conan packages even for glib.

Reply:
Sadly OBS is not among the conan officially supported CI platforms, details in https://docs.conan.io/en/latest/integrations/ci.html